### PR TITLE
fix(amazon/serverGroup): Enable launch templates for rolling push

### PR DESCRIPTION
**Pull request by caseyhebebrand**
- specifies `setLaunchTemplate` when rolling push strategy is selected by user
- adds a settings to conditionally use `launch templates` in help text depending on whether or not launch templates or launch configurations are activated.

### CODE REVIEW

File: app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts [hunk 1]
```diff
@@ -166,6 +166,8 @@ export class AwsServerGroupConfigurationService {
       command.suspendedProcesses.includes(process);
 
     cmd.onStrategyChange = (command: IAmazonServerGroupCommand, strategy: IDeploymentStrategy): void => {
+      command.setLaunchTemplate = strategy.key === 'rollingpush' ? true : undefined;
```
Comment (jeyrschabu) [inline app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts:169]: I am hoping we don't need this. It feels very specific to rollingPush.

Comment (jeyrschabu) [reply app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts:169]: The tricky thing is that rollingpush is very orca heavy. Let me play around with this branch a bit.

Comment (caseyhebebrand) [reply app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts:169]: Just let me know. For now its only useful for rollingPush, but we can always take it out if we don't end up needing this.

Comment (jeyrschabu) [reply app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts:169]: Sounds good!

File: app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts [hunk 2]
```diff
@@ -166,6 +167,10 @@ export class AwsServerGroupConfigurationService {
       command.suspendedProcesses.includes(process);
 
     cmd.onStrategyChange = (command: IAmazonServerGroupCommand, strategy: IDeploymentStrategy): void => {
+      if (AWSProviderSettings.serverGroups.enableLaunchTemplates) {
```
Comment (caseyhebebrand) [inline app/scripts/modules/amazon/src/serverGroup/configure/serverGroupConfiguration.service.ts:170]:
```suggestion
      if (AWSProviderSettings.serverGroups?.enableLaunchTemplates) {
```

**Review by jeyrschabu — APPROVED** [review]
LGTM

**Pull request merged**